tenable/io/v3/assets/api.py

```diff
@@ -0,0 +1,403 @@
+                             ("field_name_2", "desc")]``
+                    - ``[{'property': 'last_observed', 'order': 'desc'}]``
+            limit (int):
+                Number of objects to be returned in each request.
+                Default is 1000.
+            next (str):
+                The pagination token to use when requesting the next page of
+                results.  This token is presented in the previous response.
+            return_resp (bool):
+                If set to true, will override the default behavior to return
+                an iterable and will instead return the results for the
+                specific page of data.
+            return_csv (bool):
+                If set to true, It wil return the CSV Iterable. Returns all
+                data in text/csv format on each next call with row headers
+                on each page.
+
+        Returns:
+            Returns:
+                Iterable:
+                    The iterable that handles the pagination and potentially
+                    async requests for the job.
+                requests.Response:
+                    If ``return_json`` was set to ``True``, then a response
+                    object is instead returned instead of an iterable.
+
+        Examples:
+            >>> tio.v3.assets.search_assets(filter=('netbios_name', 'eq',
+            >>>  'SCCM'), fields=["name", "netbios_name", "last_login"],
+            >>>    limit=2, sort=[('last_observed', 'asc')])
+        '''
+        search_api_path = 'api/v3/assets/search'
+        return_csv = kw.pop('return_csv')
+        return super().search(
+            resource='assets',
+            iterator_cls=AssetCSVIterator if return_csv else
+            AssetSearchIterator,
+            is_sort_with_prop=False,
+            api_path=search_api_path,
+            **kw
+        )
+
+    def delete(self, uuid: UUID) -> None:
+        '''
+        Deletes the asset.
+
+        :devportal:`assets: asset-delete <asset-delete>`
+
+        Args:
+            uuid (str): The unique identifier for the asset.
+
+        Returns:
+            None:
+
+        Examples:
+            >>> asset_id = '00000000-0000-0000-0000-000000000000'
+            >>> tio.v3.delete(asset_id)
+        '''
+        return self._delete(uuid)
+
+    def details(self, uuid: UUID) -> Dict:
+        '''
+        Retrieves the details about a specific asset.
+
+        :devportal:`assets: asset-info <assets-asset-info>`
+
+        Args:
+            uuid:
+                The UUID (unique identifier) for the asset.
+
+        Returns:
+            Dict:
+                Asset resource definition.
+
+        Examples:
+            >>> asset = tio.v3.assets.details(
+            ...     '00000000-0000-0000-0000-000000000000')
+        '''
+        return self._get(f'{uuid}')
+
+    def assign_tags(
+            self, action: Literal['add', 'remove'],
+            assets: List[UUID], tags: List[UUID]
+    ) -> Dict:
```

Comment:
currently is an anti-pattern for python.  the following below are common.  we typically prefer the first one in the mainline codebase.


```suggestion
    def assign_tags(self, 
                    action: Literal['add', 'remove'],
                    assets: List[UUID], 
                    tags: List[UUID]
                    ) -> Dict:
```

or

```suggestion
    def assign_tags(
        self, 
        action: Literal['add', 'remove'],
        assets: List[UUID], 
        tags: List[UUID]
    ) -> Dict:
```